daemon/run.go

```diff
@@ -0,0 +1,319 @@
+package daemon
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/gorilla/mux"
+
+	"github.com/akitasoftware/akita-cli/apispec"
+	"github.com/akitasoftware/akita-cli/har_loader"
+	"github.com/akitasoftware/akita-cli/plugin"
+	"github.com/akitasoftware/akita-cli/rest"
+	"github.com/akitasoftware/akita-cli/trace"
+	"github.com/akitasoftware/akita-cli/util"
+	"github.com/akitasoftware/akita-libs/akid"
+	kgxapi "github.com/akitasoftware/akita-libs/api_schema"
+	"github.com/akitasoftware/akita-libs/sampled_err"
+)
+
+const WITNESS_BUFFER_SIZE = 10_000
+
+type Witness = har_loader.CustomHAREntry
+
+type Args struct {
+	// Required args.
+	ClientID akid.ClientID
+	Domain   string
+
+	// Optional args.
+	PortNumber uint16
+
+	Plugins []plugin.AkitaPlugin
+}
+
+// A wrapper around HTTP request handlers for handling HTTP responses generically.
+type httpHandler func(*http.Request) HTTPResponse
+
+var cmdArgs Args
+
+func Run(args Args) error {
+	cmdArgs = args
+	router := mux.NewRouter().StrictSlash(true)
+
+	// Register an endpoint for creating a new learning session.
+	router.Handle("/services/{serviceName}/learn/sessions", httpHandler(createLearnSession)).Methods("POST")
+
+	// Register an endpoint for adding witnesses to a learning session. The request body is expected to be a stream of HAR entry objects to be added.
+	router.Handle("/services/{serviceName}/learn/sessions/{learnSessionName}/witnesses", httpHandler(addWitnesses)).Methods("POST")
```

Comment:
Can you preface each of the paths with `/v1`, e.g. `/v1/services/...`?  We've adopted that style of versioning elsewhere, and it will make it easier to introduce new versions in the future.

Reply:
Done in 2ed6419.